### PR TITLE
feat(codepipeline): introduce the Action abstract class

### PR DESCRIPTION
**Pull request by skinny85**
Add an officially supported class to the API of the CodePipeline module
that contains some logic common to all implementations of the `IAction` interface,
like dealing with variable namespaces, or methods like `onStateChange()`.

There was previously a class like that in the codepipeline-actions module,
but it was marked experimental, and had some strong opinions on the construction way of the subclasses
(required passing the `actionProperties` through a `super()` call in the constructor,
which is not very flexible).

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: allowed-breaking-changes.txt [hunk 1]
```diff
@@ -61,3 +61,26 @@ weakened:@aws-cdk/cloud-assembly-schema.FileSource
 # These are fine, since they shouldn't be widely used.
 weakened:@aws-cdk/core.FileAssetLocation
 weakened:@aws-cdk/aws-events.RuleTargetConfig
+
+# Changes needed for moving the Action class from the codepipeline-actions
+# to the codepipeline module
+base-types:@aws-cdk/aws-codepipeline-actions.AlexaSkillDeployAction
```
Comment (rix0rrr) [inline allowed-breaking-changes.txt:67]: Fairly sure this is not okay for Java users.

Can you not leave `export class Action extends codepipeline.Action { }` in the `codepipeline-actions` module?

Comment (rix0rrr) [reply allowed-breaking-changes.txt:67]: We need to `@deprecated` it, and then probably also add it the strip-deprecated exclude list (if that already exists).

**Review by rix0rrr — CHANGES_REQUESTED** [review]

File: allowed-breaking-changes.txt (hunk 1 above)
Comment (skinny85) [reply allowed-breaking-changes.txt]: Ok. Left the "old" `Action` from the `codepipeline-actions` module in place.

There's no exclude list for deprecations, so made it 'stable' instead.

**Comment by skinny85** [comment]
@rix0rrr incorporated your comments, would appreciate a re-review!

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**